core/domain/feedback_services.py

```diff
@@ -368,7 +421,119 @@ def _get_thread_from_model(thread_model):
+                thread.
+            - 'last_message_read': boolean. Whether the last message is read by
+                the user.
+            - 'second_last_message_read': boolean. Whether the second last
+                message is read by the user,
+            - 'author_last_message': str. The name of the author of the last
+                message.
+            - 'author_second_last_message': str. The name of the author of the
+                second last message.
+            - 'exploration_title': str. The title of the exploration to which
+                exploration belongs.
+    """
+    exploration_ids, thread_ids = (
+        feedback_models.FeedbackThreadModel.get_exploration_and_thread_ids(
+            full_thread_ids))
+
+    thread_model_ids = (
+        [feedback_models.FeedbackThreadModel.generate_full_thread_id(
+            exploration_id, thread_id)
+         for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+
+    feedback_thread_user_model_ids = (
+        [feedback_models.FeedbackThreadUserModel.generate_full_id(
+            user_id, exploration_id, thread_id)
+         for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+
+    multiple_models = (
+        datastore_services.fetch_multiple_entities_by_ids_and_models(
+            [
+                ('FeedbackThreadModel', thread_model_ids),
+                ('FeedbackThreadUserModel', feedback_thread_user_model_ids),
+                ('ExplorationModel', exploration_ids),
+            ]))
+
+    thread_models = multiple_models[0]
+    feedback_thread_user_models = multiple_models[1]
+    explorations = multiple_models[2]
+
+    threads = [_get_thread_from_model(thread_model)
+               for thread_model in thread_models]
+
+    last_two_messages_ids = []
+    for thread in threads:
+        last_two_messages_ids = (
+            last_two_messages_ids + thread.get_last_two_message_ids())
+
+    messages = feedback_models.FeedbackMessageModel.get_multi(
+        last_two_messages_ids)
+
+    last_two_messages = [messages[i:i + 2] for i in range(0, len(messages), 2)]
+
+    thread_summaries = []
+    for index, thread in enumerate(threads):
+        last_message_read = (
+            last_two_messages[index][0].message_id
+            in feedback_thread_user_models[index].message_ids_read_by_user)
+        author_last_message = last_two_messages[index][0].author_id
+
+        second_last_message_read = None
+        author_second_last_message = None
+
+        does_second_message_exist = (last_two_messages[index][1] is not None)
+        if does_second_message_exist:
+            second_last_message_read = (
+                last_two_messages[index][1].message_id
+                in feedback_thread_user_models[index].message_ids_read_by_user)
+            author_second_last_message = last_two_messages[index][1].author_id
+
+        if thread.message_count:
+            total_no_of_messages = thread.message_count
+        else:
```

Comment:
(I'm having trouble finding the original comment, but I'm pretty sure we discussed this.)